hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -66,6 +66,7 @@ func filterMutableHostedClusterSpecFields(spec *hyperv1.HostedClusterSpec) {
 	}
 	if spec.Platform.Type == hyperv1.AWSPlatform && spec.Platform.AWS != nil {
 		spec.Platform.AWS.ResourceTags = nil
+		spec.Platform.AWS.RolesRef = hyperv1.AWSRolesRef{}
```

Comment:
You also need to clear the other deprecated fields because in the fixup you change those as well